pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -616,26 +617,48 @@ func (jm *ControllerV2) cleanupFinishedJobs(cj *batchv1beta1.CronJob, js []batch
 	}
 
 	if cj.Spec.SuccessfulJobsHistoryLimit != nil {
-		removeOldestJobs(cj,
+		jm.removeOldestJobs(cj,
 			successfulJobs,
-			jm.jobControl,
-			*cj.Spec.SuccessfulJobsHistoryLimit,
-			jm.recorder)
+			*cj.Spec.SuccessfulJobsHistoryLimit)
 	}
 
 	if cj.Spec.FailedJobsHistoryLimit != nil {
-		removeOldestJobs(cj,
+		jm.removeOldestJobs(cj,
 			failedJobs,
-			jm.jobControl,
-			*cj.Spec.FailedJobsHistoryLimit,
-			jm.recorder)
+			*cj.Spec.FailedJobsHistoryLimit)
 	}
 
 	// Update the CronJob, in case jobs were removed from the list.
 	_, err := jm.cronJobControl.UpdateStatus(cj)
 	return err
 }
 
+func (jm *ControllerV2) getFinishedStatus(j *batchv1.Job) (bool, batchv1.JobConditionType) {
```

Comment:
There's no need for duplicating this method, identical already exists in utils.go

Reply:
@soltysh I needed to send an instance of jm so I could call the other method [here](https://github.com/kubernetes/kubernetes/pull/96443/files#diff-3ac32b4c4bbfd547f2224dda02e1b422629791f723459a6cfb6a1a0670519975R619)

given that, I wonder if I keep this as is?

Reply:
That should not be related, but I'm fine with this as is for now.